src/pke/unittest/utckksrns/UnitTestSchemeSwitch.cpp

```diff
@@ -320,18 +320,9 @@ class UTCKKSRNS_SCHEMESWITCH : public ::testing::TestWithParam<TEST_CASE_UTCKKSR
             auto ccLWE          = cc->GetBinCCForSchemeSwitch();
             cc->EvalCKKStoFHEWKeyGen(keyPair, privateKeyFHEW);
 
-            const auto cryptoParams = std::dynamic_pointer_cast<CryptoParametersCKKSRNS>(cc->GetCryptoParameters());
-            ILDCRTParams<Element::Integer> elementParams = *(cryptoParams->GetElementParams());
-            auto paramsQ                                 = elementParams.GetParams();
-            auto modulus_CKKS_from                       = paramsQ[0]->GetModulus();
-            auto modulus_LWE                             = 1 << testData.logQ;
-            auto pLWE                                    = modulus_LWE / (2 * ccLWE->GetBeta().ConvertToInt());
-
-            double scFactor = cryptoParams->GetScalingFactorReal(0);
-            if (cryptoParams->GetScalingTechnique() == FLEXIBLEAUTOEXT)
-                scFactor = cryptoParams->GetScalingFactorReal(1);
-            double scale = modulus_CKKS_from.ConvertToInt() / (scFactor * pLWE);
-
+            auto modulus_LWE = 1 << testData.logQ;
+            auto pLWE        = modulus_LWE / (2 * ccLWE->GetBeta().ConvertToInt());
+            double scale     = 1.0 / pLWE;
```

Comment:
is the check "if (pLWE != 0)" required here?

Reply:
No, it is not. modulus_LWE is obtained as a power of two, so it can never be 0. pLWE is modulus_LWE over something always positive.